tests/unit/s2n_psk_test.c

```diff
@@ -115,6 +115,48 @@ int main(int argc, char **argv)
         EXPECT_BYTEARRAY_EQUAL(psk.secret.data, TEST_VALUE_2, sizeof(TEST_VALUE_2));
     }
 
+    /* Test s2n_psk_clone */
+    {
+        const uint8_t test_bad_value[] = "wrong";
+        const uint8_t test_identity[] = "identity";
+        const uint8_t test_secret[] = "secret";
+        const uint8_t test_early_secret[] = "early_secret";
+        const s2n_hmac_algorithm test_hmac = S2N_HMAC_SHA384;
+
+        struct s2n_psk *original = s2n_external_psk_new();
+        EXPECT_NOT_NULL(original);
+        EXPECT_SUCCESS(s2n_psk_set_identity(original, test_identity, sizeof(test_identity)));
+        EXPECT_SUCCESS(s2n_psk_set_secret(original, test_secret, sizeof(test_secret)));
+        EXPECT_SUCCESS(s2n_alloc(&original->early_secret, sizeof(test_early_secret)));
+        EXPECT_MEMCPY_SUCCESS(original->early_secret.data, test_early_secret, original->early_secret.size);
+        original->hmac_alg = test_hmac;
+
+        DEFER_CLEANUP(struct s2n_psk *clone = s2n_external_psk_new(), s2n_psk_free);
+        EXPECT_SUCCESS(s2n_psk_set_identity(clone, test_bad_value, sizeof(test_bad_value)));
+        EXPECT_SUCCESS(s2n_psk_set_secret(clone, test_bad_value, sizeof(test_bad_value)));
+        EXPECT_NOT_NULL(clone);
+
+        EXPECT_OK(s2n_psk_clone(clone, original));
+
+        /* Free the original to ensure they share no memory */
+        EXPECT_SUCCESS(s2n_psk_free(&original));
```

Comment:
Same comment here as on the other clone test. I'm worried that there are too many implicit assumptions behind this test.